File: 0739-daily-temperatures/0739-daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        int n=temperatures.size();
        vector<int> ans(n);
        stack<int> st;// push index

        for(int i=0;i<n;i++){
            while(!st.empty() && temperatures[st.top()]<temperatures[i]){
                int top=st.top();
                ans[top]=i-top;
                st.pop();
            }
            st.push(i);
        }
         while(!st.empty()){
                int top=st.top();
                ans[top]=0;
                st.pop();
            }
        return ans;
    }
};

File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n=temperatures.length;
        int arr[]=new int [n];
        Arrays.fill(arr,0);
        Stack<Integer> st=new Stack<>();
        for(int i=0;i<n;i++){
            while(!st.isEmpty() && temperatures[i]>temperatures[st.peek()]){
                arr[st.peek()]=i-st.peek();
                st.pop();
            }
            st.push(i);
        }
        return arr;
    }
}